TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -937,8 +937,11 @@ public static (SymbolType, Node) ResolveTargetSectionOrParagraph(Node callerNode
 
         private static void Check<T>(string nodeTypeName, T node, [NotNull] IList<T> found) where T : Node
         {
-            if (found.Count > 1)
+            //Check for duplicate declaration within the same owner
```

Comment:
Maybe add another comment to explain what we compare when `T` is a paragraph (Owner is either a Program or a Section) vs when `T` is a section (Owner is Program).